libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/RouterWrapper.kt

```diff
@@ -0,0 +1,227 @@
+/**
+ * Tampering with any file that contains billing code is a violation of Mapbox Terms of Service and will result in enforcement of the penalties stipulated in the ToS.
+ */
+
+package com.mapbox.navigation.route.internal
+
+import com.mapbox.annotation.module.MapboxModule
+import com.mapbox.annotation.module.MapboxModuleType
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.route.RouteRefreshCallback
+import com.mapbox.navigation.base.route.RouteRefreshError
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.RouterCallback
+import com.mapbox.navigation.base.route.RouterFailure
+import com.mapbox.navigation.route.internal.util.ACCESS_TOKEN_QUERY_PARAM
+import com.mapbox.navigation.route.internal.util.httpUrl
+import com.mapbox.navigation.route.internal.util.redactQueryParam
+import com.mapbox.navigation.route.onboard.OfflineRoute
+import com.mapbox.navigation.utils.internal.ThreadController
+import com.mapbox.navigation.utils.internal.logI
+import com.mapbox.navigation.utils.internal.logW
+import com.mapbox.navigator.RouteRefreshOptions
+import com.mapbox.navigator.RouterInterface
+import com.mapbox.navigator.RoutingProfile
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.json.JSONObject
+import java.net.URL
+
+@MapboxModule(MapboxModuleType.NavigationRouter)
+class RouterWrapper(
+    private val accessToken: String,
+    private val router: RouterInterface,
+) : Router {
+
+    private val mainJobControl by lazy { ThreadController.getMainScopeAndRootJob() }
+
+    override fun getRoute(routeOptions: RouteOptions, callback: RouterCallback): Long {
+        val httpUrl = MapboxDirections.builder()
+            .accessToken(accessToken)
+            .routeOptions(routeOptions)
+            .build()
+            .httpUrl()
```

Comment:
Noticed that https://github.com/mapbox/mapbox-navigation-android/blob/fa9cf1f19fc3e17c78fb30f7ed8e39f5cda6815e/libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/util/MapboxDirectionsEx.kt#L1 can be removed as it's `internal`.

Reply:
done